src/components/OptionList/OptionList.tsx

```diff
@@ -61,129 +64,112 @@ export interface State {
-
-    if (optionsChanged || sectionsChanged || titleChanged) {
-      this.setState({
-        normalizedOptions: createNormalizedOptions(
-          nextOptions,
-          nextSections,
-          nextTitle,
-        ),
-      });
-    }
-  }
-
-  render() {
-    const {normalizedOptions} = this.state;
-    const {selected, allowMultiple, role, optionRole} = this.props;
-    const optionsExist = normalizedOptions.length > 0;
-
-    const optionsMarkup = optionsExist
-      ? normalizedOptions.map(({title, options}, sectionIndex) => {
-          const titleMarkup = title ? (
-            <p className={styles.Title} role={role}>
-              {title}
-            </p>
-          ) : null;
-          const optionsMarkup =
-            options &&
-            options.map((option, optionIndex) => {
-              const isSelected = selected.includes(option.value);
-              const id =
-                option.id || `${this.id}-${sectionIndex}-${optionIndex}`;
-
-              return (
-                <Option
-                  {...option}
-                  key={id}
-                  id={id}
-                  section={sectionIndex}
-                  index={optionIndex}
-                  onClick={this.handleClick}
-                  select={isSelected}
-                  allowMultiple={allowMultiple}
-                  role={optionRole}
-                />
-              );
-            });
-
-          return (
-            <li key={title || `noTitle-${sectionIndex}`}>
-              {titleMarkup}
-              <ul
-                className={styles.Options}
-                id={this.id}
-                role={role}
-                aria-multiselectable={allowMultiple}
-              >
-                {optionsMarkup}
-              </ul>
-            </li>
-          );
-        })
-      : null;
-
-    return (
-      <ul className={styles.OptionList} role={role}>
-        {optionsMarkup}
-      </ul>
-    );
-  }
-
-  private handleClick = (sectionIndex: number, optionIndex: number) => {
-    const {selected, onChange, allowMultiple} = this.props;
-    const selectedValue = this.state.normalizedOptions[sectionIndex].options[
-      optionIndex
-    ].value;
-    const foundIndex = selected.indexOf(selectedValue);
-    if (allowMultiple) {
-      const newSelection =
-        foundIndex === -1
-          ? [selectedValue, ...selected]
-          : [
-              ...selected.slice(0, foundIndex),
-              ...selected.slice(foundIndex + 1, selected.length),
-            ];
-      onChange(newSelection);
-      return;
-    }
-    onChange([selectedValue]);
-  };
+export default function OptionList({
+  options,
+  sections,
+  title,
+  selected,
+  allowMultiple,
+  role,
+  optionRole,
+  onChange,
+  id: propId,
+}: Props) {
+  const [normalizedOptions, setNormalizedOptions] = React.useState(
+    createNormalizedOptions(options, sections, title),
+  );
+  const id = React.useRef(propId || getUniqueId());
+
+  React.useEffect(
```

Comment:
I separated this out from the other content that was in `componentWillReceiveProps` since it's unrelated